File: nysol_python/mcmd/methods/convTex/db/smp/func_null.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='null'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id
1
2
3
'''
db['idatas'].append(data)

data={}
data['name']='dat2.csv'
data['text']='''
id,val
1,a
2,
3,b
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
rslという項目の全行にNULL値を出力する。
'''
script['sh_code']='''
mcal c='nulls()' a=rsl i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcal(c='nulls()', a='rsl', i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='if文の中での利用'
script['comment']='''
if文の第二パラメータで数値を指定しているので、それに合わせてnulln()関数を用いる。
'''
script['sh_code']='''
mcal c='if(${id}==1,1,nulln())' a=rsl i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mcal(c='if(${id}==1,1,nulln())', a='rsl', i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='isnullと同等の指定'
script['comment']='''
'''
script['sh_code']='''
mcal c='if(${val}==nulln(),"null","notNull")' a=rsl i=dat2.csv o=rsl3.csv
'''
script['py_code']='''
nm.mcal(c='if(${val}==nulln(),"null","notNull")', a='rsl', i="dat2.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/func_null.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id
1
2
3
EOF
)}

File.open("dat2.csv","w"){|fpw| fpw.write(
<<'EOF'
id,val
1,a
2,
3,b
EOF
)}


############## Example 1
title="Basic Example"
comment=<<'EOF'
Print NULL values to column \verb|rsl|. 
EOF
scp=<<'EOF'
more dat1.csv
mcal c='nulls()' a=rsl i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Use of if statement"
comment=<<'EOF'
Use nulln() function to match the value specified in the second parameter.
EOF
scp=<<'EOF'
mcal c='if(${id}==1,1,nulln())' a=rsl i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Equivalent to isnull function"
comment=<<'EOF'
EOF
scp=<<'EOF'
mcal c='if(${val}==nulln(),"null","notNull")' a=rsl i=dat2.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)

